integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-hadoop-service-bundle/nifi-hadoop-service/src/main/java/com/thinkbiganalytics/nifi/v2/thrift/RefreshableDataSource.java

```diff
@@ -75,274 +43,60 @@
      * @param driverClassLoader the driver class loader
      * @param validationQuery   the query used to test connections
      */
-    public RefreshableDataSource(String driverClassName, String url, String username, String password,
-                                 ClassLoader driverClassLoader, String validationQuery, Long validationQueryTimeout) {
-        this.driverClassName = driverClassName;
-        this.url = url;
-        this.username = username;
-        this.password = password;
-        this.driverClassLoader = driverClassLoader;
-        this.validationQuery = validationQuery;
-        this.validationQueryTimeout = validationQueryTimeout;
-        refresh();
-    }
-
-    /**
-     * called to refresh the connection if needed
-     */
-    private void refresh() {
-        if (isRefreshing.compareAndSet(false, true)) {
-            log.info("REFRESHING DATASOURCE for {} ", this.url);
-            target.set(create());
-            isRefreshing.set(false);
-        } else {
-            //unable to refresh.  Refresh already in progress
+    public RefreshableDataSource(String driverClassName,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Constructor has 9 parameters, which is greater than 7 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00107)